### DISCUSSION TITLE
Adding Field from Query Parameters

### DISCUSSION THREAD

**Discussion by AlxGlx** [airbytehq/airbyte#38462] [Connector Questions] [answered]
I’m sure there’s a simple answer to this I can’t see because I’m low on coffee, but I have a custom connector to an API that requires a “startDate” query parameter. I would like to include this as a field in the response, using the Transformations settings. I can’t seem to get the Value correct. 

I have tried:

- {{ config["startDate"] }}
- {{ config.startDate] }}
- {{ parameters["startDate"] }}
- {{ parameters.startDate] }}

Any advice would be appreciated. Thanks.

**Comment by NAjustin** [comment, marked as answer]
I'm guessing you'll want to do this as an incremental stream, which should have the following settings:
- `API Filtering Capabilities`: "Start"
- `Inject Start Time into outgoing HTTP Request`: (checked)
- `Inject Into`: "Query Parameter"
- `Parameter Name`: `startDate`
- `Cursor Field`: (whatever the column the start date is filtering is named)
- `Cursor Datetime Formats`: (a list of whatever date formats may be included in that field)

For the initial value, you'll probably want to use the default of a user input. Then when you're grabbing the field you want to add the requested start time in Transformations, it'll be `{{ stream_interval.start_time }}` (which will be the user value on the first request, and the current incremental value on future requests)

If you're _not_ wanting the stream to be incremental and always want this to be a static value, you could just set the source value in a user input, and then use `{{ config.startDate }}` in your Transformation (assuming the user input is named `startDate`)

**Comment by AlxGlx** [comment]
Appreciate the response. Got there in the end.

I was indeed using an incremental stream, so was able to get the column added in using the `{{ stream_interval.start_time }}` as you suggest.
I still couldn’t get the column working with `{{ config.startDate }}`, which didn’t matter in the end as I used stream_interval.

Thanks for the advice.
